snuba/query/processors/type_converters/__init__.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+from typing import Set
+
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.query.conditions import FUNCTION_TO_OPERATOR, ConditionFunctions
+from snuba.query.exceptions import ValidationException
+from snuba.query.expressions import Column, Expression, FunctionCall
+from snuba.query.matchers import Any as AnyMatch
+from snuba.query.matchers import Column as ColumnMatch
+from snuba.query.matchers import FunctionCall as FunctionCallMatch
+from snuba.query.matchers import Literal as LiteralMatch
+from snuba.query.matchers import Or, Param, String
+from snuba.request.request_settings import RequestSettings
+
+
+class ColumnTypeError(ValidationException):
+    pass
+
+
+class BaseTypeConverter(QueryProcessor, ABC):
+    def __init__(self, columns: Set[str]):
+        self.columns = columns
+        column_match = Or([String(col) for col in columns])
+
+        literal = Param("literal", LiteralMatch(AnyMatch(str)))
+
+        operator = Param(
+            "operator",
+            Or(
+                [
+                    String(op)
+                    for op in FUNCTION_TO_OPERATOR
+                    if op not in (ConditionFunctions.IN, ConditionFunctions.NOT_IN)
+                ]
+            ),
+        )
+
+        in_operators = Param(
+            "operator",
+            Or((String(ConditionFunctions.IN), String(ConditionFunctions.NOT_IN))),
+        )
+
+        col = Param("col", ColumnMatch(None, column_match))
+
+        self.__condition_matcher = Or(
+            [
+                FunctionCallMatch(operator, (literal, col)),
+                FunctionCallMatch(operator, (col, literal)),
+            ]
+        )
+
+        self.__in_condition_matcher = FunctionCallMatch(
+            in_operators,
+            (col, Param("tuple", FunctionCallMatch(String("tuple"), None)),),
+        )
+
+    def strip_column_alias(self, exp: Expression) -> Expression:
+        assert isinstance(exp, Column)
+        return Column(
+            alias=None, table_name=exp.table_name, column_name=exp.column_name
+        )
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        query.transform_expressions(
+            self.process_expressions, skip_transform_condition=True
+        )
+
+        condition = query.get_condition()
+        if condition:
```

Comment:
nit: https://www.python.org/dev/peps/pep-0008/#programming-recommendations
`if condition is not None:`